src/main/java/com/programmers/domain/voucher/DiscountAmount.java

```diff
@@ -0,0 +1,32 @@
+package com.programmers.domain.voucher;
+
+import com.programmers.exception.AmountValueException;
+
+public class DiscountAmount {
+
+    private final long MIN_AMOUNT = 1;
+    private final long MAX_AMOUNT = Integer.MAX_VALUE;
+
+    private final long amount;
+
+    public DiscountAmount(long amount) {
+        validateDiscountAmount(amount);
+        this.amount = amount;
+    }
+
+    public long getAmount() {
+        return amount;
+    }
+
+    private void validateDiscountAmount(long amount) {
+        if (amount < MIN_AMOUNT || amount > MAX_AMOUNT) {
+            throw new AmountValueException();
+        }
+    }
+
+    public long discount(long beforeDiscount) {
+        return beforeDiscount - amount;
+    }
```

Comment:
값 객체를 만들어 사용하셨는데, long 타입을 사용하지 않고 `DiscountAmount`를 사용할 수 있을 것 같아요.
```suggestion
    public DiscountAmount discount(DiscountAmount beforeDiscount) {
        return new DiscountAmount(beforeDiscount.amount - amount);
    }
```